pkg/storage/kubernetes/ipam.go

```diff
@@ -31,17 +32,27 @@ import (
 // NewKubernetesIPAM returns a new KubernetesIPAM Client configured to a kubernetes CRD backend
 func NewKubernetesIPAM(containerID string, ipamConf whereaboutstypes.IPAMConfig) (*KubernetesIPAM, error) {
 	var namespace string
-	if cfg, err := clientcmd.LoadFromFile(ipamConf.Kubernetes.KubeConfigPath); err != nil {
-		return nil, err
-	} else if ctx, ok := cfg.Contexts[cfg.CurrentContext]; ok && ctx != nil {
-		namespace = ctx.Namespace
+	var kubernetesClient *Client
+	if ipamConf.Kubernetes.KubeConfigPath == "" {
+		var err error
+		kubernetesClient, err = NewClient(storage.RequestTimeout)
+		if err != nil {
+			return nil, fmt.Errorf("failed instantiating kubernetes client: %v", err)
+		}
+		namespace = metav1.NamespaceSystem
```

Comment:
this is wrong imo. See https://github.com/k8snetworkplumbingwg/whereabouts/pull/221#discussion_r870483371 .